include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -277,27 +112,55 @@ namespace seqan3
  * assert(letter.to_rank() == 6);
  * ```
  */
-template <typename first_alphabet_type, typename ...alphabet_types>
+template <typename ...alphabet_types>
 //!\cond
-    requires alphabet_concept<first_alphabet_type> && (alphabet_concept<alphabet_types> && ...)
+    requires (alphabet_concept<alphabet_types> && ... && (sizeof...(alphabet_types) >= 1))
 //!\endcond
 class union_composition
 {
+protected:
+    //!\privatesection
+
+    /*!\brief Returns the sum over all alphabet_size_v<alphabets_t>'s.
+     * \tparam ...alphabets_t The types must satisfy seqan3::alphabet_concept.
+     * \hideinitializer
+     *
+     * ```cpp
+     * constexpr auto sum = alphabet_sum_size_v<dna4, gap, dna5>;
+     * assert(sum == 10);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto alphabet_sum_size_v =
+        (static_cast<size_t>(alphabet_size_v<alphabets_t>) + ... + static_cast<size_t>(0));
+
+    //!\publicsection
 public:
     /*!\brief Returns true if alphabet_t is one of the given alphabet types.
      * \tparam alphabet_t The type to check
+     *
+     * ```cpp
+     * using union_t = union_composition<dna4, gap>;
+     *
+     * static_assert(union_t::has_type<dna4>(), "should be true");
+     * static_assert(union_t::has_type<gap>(), "should be true");
+     * static_assert(!union_t::has_type<dna5>(), "should be false");
+     * ```
      */
     template <typename alphabet_t>
     static constexpr bool has_type() noexcept
```

Comment:
True!

Maybe `has_alternative` (has_type) and `holds_alternative` (current type)? [the "alternative" name is not great, but it would add consistency]

What do you think about the other parts of the std::variant interface like std::get? [this would be another PR then ;-) ]